### DISCUSSION TITLE
Kong Prometheus: consumer label

### DISCUSSION THREAD

**Discussion by mgrazianoc** [Kong/kong#11174] [Help] [answered]
Is there an way to configure kong prometheus plugin to include the consumer label when exporting kong's metrics? I've only seen them available (Kong 3.3.x) with `kong_http_requests_total` and `kong_bandwidth_bytes`, but it'd also want it exported (for example) with `kong_request_latency_ms_bucket` and others.

**Comment by hbagdi** [comment, marked as answer]
No, that is not possible. We don't do this out of the box because the cardinality of latency metrics with consumers (in fact all metrics) will be very high and negatively impact the performance of Prometheus and Kong significantly.
You have to fork the plugin to do this and I advise you against doing this unless you have a few 10s or 100s of consumers.
A log analysis engine is a better approach out. Prometheus is not an API analytics tool, it is an observability tool so we have no plans to enhance this further.

**Comment by mgrazianoc** [comment]
I see, thanks for the answer. We do have less than 100 consumers, they are aggregated by thousand of users or more.

In our case, depending on the consumer, we're gonna have different request durations, due to the operations it causes at the end. So, it does not fit into the analytic use case, but only observability, the same way we do for `kong_http_requests_total`.

I'm glad to know it's possible to fork it and adapt to our needs.
Nevertheless, I'm gonna keep an eye on the performance aspect
